include/ChocAn/app/state_viewer.hpp

```diff
@@ -30,7 +31,10 @@ class State_Viewer
 
     virtual ~State_Viewer() {}
 
+    virtual void update() = 0;
     virtual void render_state(const Application_State& state) = 0;
+    virtual void render_prompt(const std::string& prompt) = 0;
```

Comment:
render prompt and render user exception should also be unnecessary. They were created during the first draft of the add transaction feature.

Reply:
ill wait and see where your going with this then, since i also see your last commit was not long ago and your likely working on the branch right now.